src/components/test-slot/candidate-link/candidate-link.ts

```diff
@@ -30,15 +37,85 @@ export class CandidateLinkComponent {
   @Input()
   testComplete: boolean;
 
-  @Input()
   isPortrait: boolean;
 
+  truncatedName: string;
+
   constructor(
+    private cd: ChangeDetectorRef,
+    private ellipses: EllipsisPipe,
     public modalController: ModalController,
     public app: AppComponent,
   ) {
   }
 
+  async ngOnInit() {
+    this.truncatedName = this.getName();
+    await this.monitorOrientation();
+  }
+
+  truncateValue() {
+    let finalSum = 0;
+
+    switch (this.app.getTextZoomClass()) {
+      case 'text-zoom-regular':
+        finalSum += 35;
+        if (!this.isPortrait) {
+          finalSum += 13;
+        }
```

Comment:
Can we make these a little nicer on the eye

`finalSum += (this.isPortrait ? 35 : 13);`